### PR TITLE
chore: Update readme

### LINKED ISSUE #12: Add "how to use" in README
How to declare a dependency on this sdk in my own project? Most sdk's have a copy-paste to get you most of the way there.

### PR DESCRIPTION
**Pull request by poppoerika**
Note:
- Need to update the example link once https://github.com/momentohq/client-sdk-examples/pull/69 is merged

Closes #12

### CODE REVIEW

File: .github/workflows/manual_release.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          set -e
+          set -x
+          export MOMENTO_SDK_VERSION="${{needs.release.outputs.version}}"
+          if [ -z "$MOMENTO_SDK_VERSION"]
+          then
+            echo "Unable to determine SDK version!  Exiting!"
+            exit 1
+          fi
+          echo "MOMENTO_SDK_VERSION=${MOMENTO_SDK_VERSION}"
+          GOPROXY=proxy.golang.org go list -m example.com/mymodule@v${MOMENTO_SDK_VERSION}
```
Comment (eaddingtonwhite) [inline .github/workflows/manual_release.yml:80]: is this right?

Comment (poppoerika) [reply .github/workflows/manual_release.yml:80]: 😱

File: .github/workflows/manual_release.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          set -e
+          set -x
+          export MOMENTO_SDK_VERSION="${{needs.release.outputs.version}}"
+          if [ -z "$MOMENTO_SDK_VERSION"]
+          then
+            echo "Unable to determine SDK version!  Exiting!"
+            exit 1
+          fi
+          echo "MOMENTO_SDK_VERSION=${MOMENTO_SDK_VERSION}"
+          GOPROXY=proxy.golang.org go list -m github.com/momentohq/client-sdk-go@v${MOMENTO_SDK_VERSION}
```
Comment (eaddingtonwhite) [inline .github/workflows/manual_release.yml:80]: what does this step do exactly? just ensure you can find version and warms cache in central goproxy? im newer to goproxy have not used before.

File: README.md [hunk 3]
```diff
@@ -1,13 +1,105 @@
...
+	key := []byte(uuid.NewString())
+	value := []byte(uuid.NewString())
+	fmt.Printf("Setting key: %s, value: %s", key, value)
+	_, err = client.Set(&momento.CacheSetRequest{
+		CacheName: CacheName,
+		Key:       key,
+		Value:     value,
+	})
+	if err != nil {
+		fmt.Println(err.Error())
```
Comment (eaddingtonwhite) [inline README.md:81]: ditto feed back here as in example repo as well about error handling.

File: .github/workflows/manual_release.yml (hunk 2 above)
Comment (poppoerika) [reply .github/workflows/manual_release.yml:80]: This step is to publish the updated package to https://pkg.go.dev/
I referenced these:
https://go.dev/doc/modules/publishing
https://pkg.go.dev/about

File: .github/workflows/manual_release.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Install goimports
+        run: go get golang.org/x/tools/cmd/goimports
+
+      - run: goimports -l -w ./internal/grpcmanagers ./internal/models ./internal/interceptor ./internal/momentoerrors ./internal/resolver ./internal/services ./internal/utility ./momento
```
Comment (eaddingtonwhite) [inline .github/workflows/manual_release.yml:46]: hmm this seems hard to maintain. is there anyway to pass wild card here? or pass sub bash command like `$(ls -l)? We can iterate in follow up pr though.

File: README.md [hunk 5]
```diff
@@ -1,13 +1,105 @@
...
-# Prerequisites
+Go SDK for Momento, a serverless cache that automatically scales without any of the operational overhead required by traditional caching solutions.
+
+<br/>
+
+# Getting Started :running:
+
+## Requirements
+
+- [Go version 0.17.\*](https://go.dev/dl/)
```
Comment (danielamiao) [inline README.md:25]: Same comment here re: version should be `1.17.*`

File: .github/workflows/manual_release.yml (hunk 4 above)
Comment (eaddingtonwhite) [reply .github/workflows/manual_release.yml:46]: cant we do same as we are doing in ci build here?

File: README.md [hunk 6]
```diff
@@ -1,13 +1,105 @@
...
+	)
+
+	// Initializes Momento
+	client, err := momento.SimpleCacheClient(&momento.SimpleCacheClientRequest{
+		AuthToken:         AuthToken,
+		DefaultTtlSeconds: ItemDefaultTtlSeconds,
+	})
+	if err != nil {
+		panic(err)
+	} else {
```
Comment (eaddingtonwhite) [inline README.md:61]: can drop the else here.

File: README.md [hunk 7]
```diff
@@ -1,13 +1,105 @@
+# Momento client-sdk-go
+
+:warning: Experimental SDK :warning:
```
Comment (eaddingtonwhite) [inline README.md:3]: we should add some badges here in follow up pr :)
https://pkg.go.dev/badge/

File: .github/workflows/manual_release.yml (hunk 4 above)
Comment (poppoerika) [reply .github/workflows/manual_release.yml]: Forgot to update...

File: README.md [hunk 8]
```diff
@@ -1,13 +1,104 @@
...
+	if err != nil {
+		panic(err)
+	}
+    // Create Cache and check if CacheName exists
+    err := client.CreateCache(&momento.CreateCacheRequest{
+        CacheName: CacheName,
+    })
+    if err != nil {
+        panic(err)
+    }
```
Comment (danielamiao) [inline README.md:68]: weird indenting

File: README.md [hunk 9]
```diff
@@ -1,13 +1,104 @@
...
+	log.Printf("Getting key: %s\n", key)
+	resp, err := client.Get(&momento.CacheGetRequest{
+		CacheName: CacheName,
+		Key:       key,
+	})
+	if err != nil {
+		panic(err)
+	}
+    log.Printf("Lookup resulted in a : %s\n", resp.Result())
+    log.Printf("Looked up value: %s\n", resp.StringValue())
```
Comment (danielamiao) [inline README.md:92]: indent is off here too

**Review by danielamiao — APPROVED** [review]
Did you figure out why the linting step had to be moved to a separate job?

**Comment by poppoerika** [comment]
@danielamiao I'm not entirely sure what it means but I think it's because of [this](https://github.com/golangci/golangci-lint-action/issues/23#issuecomment-647177770).

**Review by danielamiao — APPROVED** [review]

**Pull request merged**